app/models/invitation.rb

```diff
@@ -60,12 +60,18 @@ def sent?
   def notify_invitee
     # TODO: maybe subclass to Invitation and ProviderInvitation
     if account.provider?
-      ProviderInvitationMailer.invitation(self).deliver_now
+      ProviderInvitationMailer.register_observer(InvitationDeliveryObserver)
+      ProviderInvitationMailer.invitation(self).deliver_later
     else
-      InvitationMailer.invitation(self).deliver_now
+      InvitationMailer.register_observer(InvitationDeliveryObserver)
+      InvitationMailer.invitation(self).deliver_later
```

Comment:
It is a pity we are still on 5.0 because we could use another Job then ActionMailer::DeliveryJob to do the deilvery in background

Instead of using deliver_later, I would create (temporarily until we upgrade) another mailer job that will do the `deliver_now` and set the `sent_at` attribute